comp/forwarder/defaultforwarder/blocked_endpoints.go

```diff
@@ -67,15 +121,48 @@ func (e *blockedEndpoints) close(endpoint string) {
 	if knownBlock, ok := e.errorPerEndpoint[endpoint]; ok {
 		b = knownBlock
 	} else {
-		b = &block{}
+		b = &block{
+			endpoint: endpoint,
+			state:    unblocked,
+		}
 	}
 
-	b.nbError = e.backoffPolicy.IncError(b.nbError)
-	b.until = time.Now().Add(e.getBackoffDuration(b.nbError))
+	b.m.RLock()
```

Comment:
This lock is redundant, because of the `e.m.Lock()` above - none of the other methods here will be accessing blocks concurrently.